huaweicloud/services/ces/resource_huaweicloud_ces_alarm_template.go

```diff
@@ -0,0 +1,381 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product CES
+// ---------------------------------------------------------------
+
+package ces
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceCesAlarmTemplate() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCesAlarmTemplateCreate,
+		UpdateContext: resourceCesAlarmTemplateUpdate,
+		ReadContext:   resourceCesAlarmTemplateRead,
+		DeleteContext: resourceCesAlarmTemplateDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the name of the CES alarm template.`,
+			},
+			"policies": {
+				Type:        schema.TypeList,
+				Elem:        AlarmTemplatePolicySchema(),
+				Required:    true,
+				Description: `Specifies the policy list of the CES alarm template.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the description of the CES alarm template.`,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the type of the CES alarm template.`,
+			},
+			"association_alarm_total": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Indicates the total num of the alarm that associated with the alarm template.`,
+			},
+			"delete_associate_alarm": {
```

Comment:
missing in doc, and it looks this should be a parameter used in delete func

Reply:
done